File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/ApplicationRoot.java
package com.github.grzesiek_galezowski.BastardInjection;

import com.github.grzesiek_galezowski.BastardInjection.Core.TeleComSystem;

import java.io.IOException;

public class ApplicationRoot {
  public static void main(String[] args) throws IOException {
    TeleComSystem sys = new TeleComSystem(); //bastard injected - look at output socket which is disposable in this example!
    sys.start();

    sys.close(); //!!!!!!!!!!!!!!!!!!!!!!!!!!!! error!!!!
  }
}

File: CSharp/DependencyInjection/DependencyInjection/BastardInjection/_4_BastardInjection/ApplicationRoot.cs
using BastardInjection._4_BastardInjection.Core;

namespace BastardInjection._4_BastardInjection
{
    public class ApplicationRoot
    {
      public static void Main(string[] args)
      {
        var sys = new TeleComSystem(); //bastard injected - look at output socket which is disposable in this example!
        sys.Start();

        sys.Dispose(); //!!!!!!!!!!!!!!!!!!!!!!!!!!!! error!!!! 
      }
    }




}
